metron-platform/metron-common/src/main/java/org/apache/metron/common/hadoop/SequenceFileIterable.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.hadoop.io.SequenceFile;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static java.lang.String.format;
+
+public class SequenceFileIterable implements Iterable<byte[]> {
+  private static final Logger LOGGER = Logger.getLogger(SequenceFileIterable.class);
+  private List<Path> files;
+  private Configuration config;
+
+  public SequenceFileIterable(List<Path> files, Configuration config) {
+    this.files = files;
+    this.config = config;
+  }
+
+  @Override
+  public Iterator<byte[]> iterator() {
+    return Iterators.concat(getIterators(files, config));
+  }
+
+  private Iterator<byte[]>[] getIterators(List<Path> files, Configuration config) {
+    return files.stream().map(f -> new SequenceFileIterator(f, config)).toArray(Iterator[]::new);
+  }
+
+  /**
+   * Cleans up all files read by this Iterable
+   *
+   * @return true if success, false if any files were not deleted
+   * @throws IOException
+   */
+  public boolean cleanup() throws IOException {
+    FileSystem fileSystem = FileSystem.get(config);
+    boolean success = true;
+    for (Path file : files) {
+      success &= fileSystem.delete(file, false);
+    }
+    return success;
+  }
+
+  private static class SequenceFileIterator implements Iterator<byte[]> {
+    private Path path;
+    private Configuration config;
+    private SequenceFile.Reader reader;
+    private LongWritable key = new LongWritable();
+    private BytesWritable value = new BytesWritable();
+    private byte[] next;
+    private boolean finished = false;
+
+    public SequenceFileIterator(Path path, Configuration config) {
+      this.path = path;
+      this.config = config;
+    }
+
+    @Override
+    public boolean hasNext() {
+      if (!finished && null == reader) {
+        try {
+          reader = new SequenceFile.Reader(config, SequenceFile.Reader.file(path));
+          LOGGER.debug("Writing file: " + path.toString());
+        } catch (IOException e) {
+          throw new RuntimeException("Failed to get reader", e);
+        }
+      } else {
+        LOGGER.debug(format("finished=%s, reader=%s, next=%s", finished, reader, next));
+      }
+      try {
+        //ensure hasnext is idempotent
+        if (!finished) {
+          if (null == next && reader.next(key, value)) {
+            next = value.copyBytes();
+          } else if (null == next) {
+            close();
+          }
+        }
+      } catch (IOException e) {
+        close();
+        throw new RuntimeException("Failed to get next record", e);
+      }
+      return (null != next);
+    }
+
+    private void close() {
+      LOGGER.debug("Closing file: " + path.toString());
+      finished = true;
+      try {
+        if (reader != null) {
+          reader.close();
+          reader = null;
+        }
+      } catch (IOException e) {
+        // ah well, we tried...
```

Comment:
Can we log this Exception?  I'm not sure we can do anything about it, but it would be nice to be able to ensure we can see it.